sof_wrapper/api/fhir.py

```diff
@@ -246,7 +246,8 @@ def route_fhir(relative_path, session_id):
     current_app.logger.debug('received session_id as path parameter: %s', session_id)
 
     session_data = get_redis_session_data(session_id)
-    patient_id = session_data.get('token_response', {}).get('patient')
+    # prefer patient ID baked into access token JWT by EHR; fallback to initial launch token for fEMR
+    patient_id = session_data.get('token_response', {}).get('patient') or session_data.get('launch_token_patient')
```

Comment:
I'd be tempted to add a check should they both exist and not match - i.e. logger.warn?

Reply:
According to the SoF spec, the launch token can be opaque. [We're copying the way SMIT encodes launch tokens](https://github.com/uwcirg/sof-api-wrapper/blob/develop/sof_wrapper/auth/views.py#L8-L24), but it's not standardized, just convenient.

Piggybacking the way SMIT encodes launch tokens (and keeping them introspectable) allows us to work around the fact that Keycloak cannot bake the patient ID into the access token (JWT).

...so in short, we probably will never see both populated because "real" EHRs will not adhere to the way SMIT encodes launch tokens

Reply:
key word, "probably" :) 

I'm not that concerned, it's just the kind of thing that leaves one scratching their heads.  Go ahead and merge when ready.